File: Python/PieceOfCake.py
'''
Problem: https://open.kattis.com/problems/pieceofcake2
Author:  Bivash Pandey
'''
length, horCut, verCut = map(int, input().split())
firstVol = 4 * horCut * verCut
secondVol = 4 * verCut * (length - horCut)
thirdVol = 4 * (length - horCut) * (length - verCut)
fourthVol = 4 * (length - verCut) * horCut
print(max(max(firstVol, secondVol), max(thirdVol, fourthVol)))

File: Java/PieceOfCake.java
/*
* Problem: https://open.kattis.com/problems/pieceofcake2
* 
* @author  Bivash Pandey
*/

import java.util.Scanner;

public class PieceOfCake{
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int length = input.nextInt();
        int horCut = input.nextInt();
        int verCut = input.nextInt();
        int firstVol = 4 * horCut * verCut;
        int secondVol = 4 * verCut * (length - horCut);
        int thirdVol = 4 * (length-horCut) * (length-verCut);
        int fourthVol = 4 * (length - verCut) * horCut;
        System.out.println(Math.max(Math.max(firstVol, secondVol), Math.max(thirdVol, fourthVol)));
    }
}